File: Code/Components/DanpheEMR.ServerModel/CommonModels/InvoiceHeaderModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel.CommonModels
{
    public class InvoiceHeaderModel
    {
        [Key]
        public int InvoiceHeaderId { get; set; }
        public string Module { get; set; }
        public string HeaderDescription { get; set; }
        public string HospitalName { get; set; }
        public string Address { get; set; }
        public string Email { get; set; }
        public string PAN { get; set; }
        public string Telephone { get; set; }
        public string DDA { get; set; }
        public string LogoFileName { get; set; }
        public string LogoFileExtention { get; set; }
        public DateTime? CreatedOn { get; set; }
        public int? CreatedBy { get; set; }
        public bool? IsActive { get; set; }

        public DateTime? ModifiedOn { get; set; }
        public int? ModifiedBy { get; set; }
        [NotMapped]
        public byte[] FileBinaryData { get; set; }
    }
}

File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/shared/invoice-header.model.ts
import { FormGroup, FormBuilder, Validators } from "@angular/forms";

export class InvoiceHeaderModel {

  public InvoiceHeaderId: number = 0;
  public Module: string = null;
  public HeaderDescription: string = null;
  public HospitalName: string = null;
  public Address: string = null;
  public Email: string = null;
  public PAN: string = null;
  public Telephone: string = null;
  public DDA: string = null;
  public LogoFileName: string = "";
  public LogoFileExtention: string = "";
  public CreatedOn: string = null;
  public CreatedBy: number = 0;
  public IsActive: boolean = true;
  public ModifiedOn: string = null;
  public ModifiedBy: number = 0;
  public FileBinaryData: string = "";


  public HeaderValidators: FormGroup = null;

  constructor() {
    var _formBuilder = new FormBuilder();
    this.HeaderValidators = _formBuilder.group({
      'HospitalName': ['', Validators.compose([Validators.required, Validators.maxLength(200)])],
      'Address': ['', Validators.compose([Validators.required, Validators.maxLength(200)])],
      'Telephone': ['', Validators.compose([Validators.required])],
      'Email': ['', Validators.compose([Validators.required, Validators.pattern('^[a-z0-9]+(\.[_a-z0-9]+)*@[a-z0-9-]+(\.[a-z0-9-]+)*(\.[a-z]{2,15})$')])],
      'PAN': ['', Validators.compose([Validators.pattern('^[a-zA-Z0-9_@./#&+-]+$')])],
      'DDA': ['', Validators.compose([Validators.maxLength(200)])],

    });
  }

  public IsDirty(fieldName): boolean {
    if (fieldName == undefined)
      return this.HeaderValidators.dirty;
    else
      return this.HeaderValidators.controls[fieldName].dirty;
  }

  public IsValid(): boolean {
    if (this.HeaderValidators.valid) { return true; } else
    { return false; }
  }
  public IsValidCheck(fieldName, validator): boolean {
    if (fieldName == undefined) {
      return this.HeaderValidators.valid;
    }
    else {
      return !(this.HeaderValidators.hasError(validator, fieldName));
    }
  }
}
